package-puzzles

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+
+import argparse
+import base64
+import glob
+import hashlib
+import hmac
+import io
+import json
+import os
+import markdown
+import random
+import sys
+import zipfile
+
+import puzzles
+
+def write_kv_pairs(ziphandle, filename, kv):
+    """ Write out a sorted map to file
+    :param ziphandle: a zipfile object
+    :param filename: The filename to write within the zipfile object
+    :param kv:  the map to write out
+    :return:
+    """
+    filehandle = io.StringIO()
```

Comment:
Why the intermediate step of a StringIO instead of writing directly to ziphandle?